### PR TITLE
[EOC-469] socket emit cohort data to proper rooms

### PR DESCRIPTION
**Pull request by alemikolo**
I've merged EOC_511 to this branch so this branch, unfortunately, includes the `saveActivitiy` refactorization also.

### CODE REVIEW

File: server/common/utils/helpers.js [hunk 1]
```diff
@@ -353,9 +354,11 @@ const responseWithCohortDetails = (doc, userId) => {
     isArchived,
     isMember: true,
     locks,
+    membersCount,
     members,
     name
   };
+  // console.log('response', ownerIds, userId);
```
Comment (marekrozmus) [inline server/common/utils/helpers.js]: commented out code

File: server/sockets/list.js [hunk 2]
```diff
@@ -353,14 +355,14 @@ const leaveList = io => async data => {
         .emit(AppEvents.LEAVE_ROOM, listMetaDataChannel(listId))
     );
 
-    socketIds.forEach(socketId =>
-      io.sockets.to(socketId).emit(ListActionTypes.DELETE_SUCCESS, data)
-    );
+    // socketIds.forEach(socketId =>
```
Comment (marekrozmus) [inline server/sockets/list.js]: commented out code

File: server/sockets/helpers/index.js [hunk 3]
```diff
@@ -336,19 +292,86 @@ const emitRemoveListViewer = (io, event) => async data => {
...
+      if (isViewer(list, userId)) {
+        listIdsUserRemained.push(listId);
+      } else if (type === ListType.SHARED) {
+        listIdsUserWasRemovedFrom.push(listId);
+      }
+    });
+
+    await Promise.all(
+      listIdsUserWasRemovedFrom.map(async listId => {
+        const data = { listId, performer, userId };
```
Comment (adamklepacz) [inline server/sockets/helpers/index.js]: line of a break

File: server/controllers/list.js [hunk 4]
```diff
@@ -1876,18 +1839,16 @@ const moveItem = async (req, resp) => {
     // prepare data to save activity and for socket
```
Comment (adamklepacz) [inline server/controllers/list.js]: remove comment

File: server/controllers/cohort.js [hunk 5]
```diff
@@ -140,112 +140,107 @@ const archiveCohort = async (req, resp) => {
...
+        ownerIds: userId
+      },
+      dataToUpdate
+    ).exec();
+    const socketInstance = io.getInstance();
+    const data = { cohortId };
+    let cohortActivity;
+
+    if (isDefined(description)) {
+      const { description: prevDescription } = cohort;
```
Comment (adamklepacz) [inline server/controllers/cohort.js]: line of a break

File: client/src/sockets/helpers/index.js [hunk 6]
```diff
@@ -315,6 +389,18 @@ export const cohortEventsController = (event, data, { dispatch, getState }) => {
         payload
       });
     }
+    case CohortEvents.FETCH_META_DATA_SUCCESS: {
+      const {
+        location: { pathname }
+      } = history;
+
```
Comment (adamklepacz) [inline client/src/sockets/helpers/index.js]: remove line of a break

**Pull request merged**